sdk/nodejs/eventgrid/zMixins.ts

```diff
@@ -143,6 +145,35 @@ export class EventGridCallbackSubscription<T> extends appservice.EventSubscripti
     }
 }
 
+async function waitUntilEndpointIsUp(resource: pulumi.Resource, url: string): Promise<string> {
+    if (pulumi.runtime.isDryRun()) {
+        return url;
+    }
+
+    // Prepare a sample webhook validation call.
+    const headers = { "aeg-event-type": "SubscriptionValidation" };
+    const body = "[{ \"data\": { \"validationCode\": \"pulumi-create\" }, \"eventType\": \"Microsoft.EventGrid.SubscriptionValidationEvent\" }]";
+
+    // Wait for up to 5 minutes
+    for (let i = 0; i < 30; i++) {
+        let status;
+        try {
+            const response = await fetch(url, { method: "POST", headers, body });
+            if (response.ok) {
+                return url;
+            }
+            status = `${response.status}: ${response.statusText}`;
+        } catch (e) {
+            status = `Error: ${e}`;
+        }
+        pulumi.log.info(`Waiting for Webhook ${url} to become available (${status}, ${i})`, resource);
+        // Wait for 10s between polls
+        await new Promise(r => setTimeout(r, 10000));
+    }
+
+    throw new Error("timed out waiting for Webhook to become up");

```

Comment:
Is "Webhook" the right word for this?

Reply:
We wait for `https://app.azurewebsites.net/runtime/webhooks/eventgrid` so I called it webhook everywhere. I'm okay to change the wording if you think it can make things more clear?